services/oauth/v2/http/transport.go

```diff
@@ -178,7 +184,11 @@ func (t *OAuthTransport) postRoundTrip(rts *roundTripState) (*http.Response, err
 				obskit.WorkspaceID(rts.destination.WorkspaceID),
 				obskit.DestinationType(rts.destination.DefinitionName),
 				logger.NewStringField("flow", string(t.flow)))
-			return nil, fmt.Errorf("getting secret from context post roundTrip")
+			// Instead of returning an error, set a 500 status code in the interceptor response
+			interceptorResp.StatusCode = http.StatusInternalServerError
+			interceptorResp.Response = "Error getting secret from context"
```

Comment:
Same as above
```suggestion
			interceptorResp.Response = "[In Oauth PostRoundTrip]Error getting secret from context"
```

Reply:
But its easier to read the error messages, and its not a lot of information overload

Reply:
where will you read this error message from? in events tab this won't be available till it has become terminal state.